src/promptflow/promptflow/executor/_service/utils/process_utils.py

```diff
@@ -0,0 +1,107 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import asyncio
+import contextlib
+import json
+import multiprocessing
+import os
+from datetime import timedelta
+from typing import Callable
+
+from promptflow._core._errors import UnexpectedError
+from promptflow._core.operation_context import OperationContext
+from promptflow._utils.exception_utils import ExceptionPresenter, JsonSerializedPromptflowException
+from promptflow._utils.logger_utils import service_logger
+from promptflow.exceptions import ErrorTarget
+from promptflow.executor._service._errors import ExecutionTimeoutError
+from promptflow.executor._service.utils.service_utils import get_log_context
+
+LONG_WAIT_TIMEOUT = timedelta(days=1).total_seconds()
+SHORT_WAIT_TIMEOUT = 10
+
+
+async def invoke_function_in_process(
```

Comment:
invoke_sync_function_in_process

Let's only support sync function now.

Reply:
Updated in another PR.
https://github.com/microsoft/promptflow/pull/2043/